feat(pipes): add intl (number and date) pipes

piloopin opened the pull request:
Closes #2340.

piloopin committed:
feat(pipes): add number (decimal, percent, currency) pipes

piloopin committed:
feat(pipes): add date pipe

vsavkin commented:
Could you split this PR into four separate PRs? The number and dates pipes are non-controversial and can be merged with a few minor changes. 

The last two commits are a bit trickier. For instance, the way you use Parser to get AST nodes, which you then call eval on, will not work with Dart transformers. We need to think a little bit more about it.

piloopin commented:
Thanks @vsavkin for the review. I addressed your comments. 

The commits are conflicting and will require manual merge when separated. I created #2956 and #2957 for filter and orderBy pipes, but kept the other two for this PR as you said they're almost good to go.

vsavkin commented:
@piloopin Thank you. I will merge this PR today.
